### PR TITLE
More authz

### LINKED ISSUE #1176: Support more Authz Features
Authz now supports smart contracts! The actions UI for Authz should support them as well! This allows DAOs to authorize certain interactions with smart contracts (including calling specific messages or allowing for migration)! 

In addition, `authz` has supported spend operations for a while now. Would be good to support those too.

### PR DESCRIPTION
**Pull request by JakeHartnell**
Closes #1176

### CODE REVIEW

File: packages/utils/messages.ts [hunk 1]
```diff
@@ -256,6 +286,97 @@ export const decodeStargateMessage = ({
...
+        },
+      })
+    }
+  } else if ('stargate' in msg) {
+    encoded = {
+      typeUrl: msg.stargate.type_url,
+      value: fromBase64(msg.stargate.value),
+    }
+  } else {
+    console.error('Msg type not supported')
```
Comment (JakeHartnell) [inline packages/utils/messages.ts:375]: Not sure what we want to do here if it's an unsupported message type. Think I covered all the options, but maybe we should throw an error?

File: packages/utils/messages.ts [hunk 2]
```diff
@@ -179,8 +196,28 @@ export const typesRegistry = new Registry([
   // Custom types not in @cosmjs/stargate default registry.
   ...([
     ['/cosmos.slashing.v1beta1.MsgUnjail', MsgUnjail],
+    ['/cosmos.authz.v1beta1.MsgGrant', MsgGrant],
+    ['/cosmos.authz.v1beta1.MsgExec', MsgExec],
     ['/cosmos.authz.v1beta1.GenericAuthorization', GenericAuthorization],
+    ['/cosmos.authz.v1beta1.Grant', Grant],
```
Comment (NoahSaso) [inline packages/utils/messages.ts:202]: Screenshot 2023-03-23 at 11 43 04 PM

I believe the 3 `authz` messages you added are included by default in the registry types from line 193. only `GenericAuthorization` needs to be manually added here from `authz`

File: packages/utils/messages.ts [hunk 3]
```diff
@@ -179,8 +196,28 @@ export const typesRegistry = new Registry([
   // Custom types not in @cosmjs/stargate default registry.
   ...([
     ['/cosmos.slashing.v1beta1.MsgUnjail', MsgUnjail],
+    ['/cosmos.authz.v1beta1.MsgGrant', MsgGrant],
+    ['/cosmos.authz.v1beta1.MsgExec', MsgExec],
```
Comment (NoahSaso) [inline packages/utils/messages.ts:200]: Screenshot 2023-03-23 at 11 43 04 PM

I believe these 2 `authz` messages are included by default in the registry types from line 193

**Comment by JakeHartnell** [comment]
Not sure why I'm getting that build error.

File: packages/utils/messages.ts (hunk 1 above)
Comment (NoahSaso) [reply packages/utils/messages.ts:378]: this fn does the same thing as `cwMsgToEncodeObject`, which we use to simulate
proposal messages :)

**Review by NoahSaso — APPROVED** [review]

**Pull request merged**